shared/wallets/wallet-list/container.js

```diff
@@ -0,0 +1,56 @@
+// @flow
+import {WalletList, type Props} from '.'
+import logger from '../../logger'
+import {connect, type TypedState, type Dispatch} from '../../util/container'
+import {getAccounts, getSelectedAccount} from '../../constants/wallets'
+import {createSelectAccount} from '../../actions/wallets-gen'
+import {type AccountID} from '../../constants/types/wallets'
+
+const mapStateToProps = (state: TypedState) => ({
+  accounts: getAccounts(state),
+  selectedAccount: getSelectedAccount(state),
+  me: state.config.username || '',
+})
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  onSelectAccount: (accountID: AccountID) => {
+    dispatch(createSelectAccount({accountID}))
+  },
+  onAddNew: () => {
+    logger.error('TODO: onAddNew')
+  },
+  onLinkExisting: () => {
+    logger.error('TODO: onLinkExisting')
+  },
+})
+
+const mergeProps = (stateProps, dispatchProps): Props => {
+  // Accounts is already ordered, so no need to sort.
+  const wallets = stateProps.accounts
+    .map(a => {
+      let name = a.name || a.accountID
+      // TODO: Better way to do this?
+      if (name.length > 20) {
```

Comment:
likely better to truncate using the component, aka how we do the chat inbox rows etc